lib/fizzy/execute.hpp

```diff
@@ -41,6 +41,28 @@ constexpr ExecutionResult Void{true};
 /// Shortcut for execution that resulted in a trap.
 constexpr ExecutionResult Trap{false};
 
+
+class ThreadContext
+{
+    class [[nodiscard]] Guard
+    {
+        ThreadContext& m_thread_context;
+
+    public:
+        explicit Guard(ThreadContext& ctx) noexcept : m_thread_context{ctx} {}
+        ~Guard() noexcept { --m_thread_context.depth; }
+    };
+
+public:
+    int depth = 0;
+
+    Guard bump_call_depth() noexcept
```

Comment:
Hm, this returns private class, I wouldn't think it's possible...

Reply:
Maybe call it `increment_call_depth`, bump sounds a bit informal it seems.

Reply:
Just an idea: it could be execute member function of `ThreadContext` bumping depth and redirecting to normal `execute`
```cpp
    ExecutionResult execute(Instance& instance, FuncIdx func_idx, const Value* args) noexcept;
```
then `Guard` can be completely hidden from user.

But maybe confusingly too many execute variants then.